### PR TITLE
IndexExchange Display Bid Adapter

### PR DESCRIPTION
**Pull request by jas-singh**
## Type of change

- [ ] Bugfix
- [ ] Feature
- [x] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change
Adding IndexExchange bid adapter

- contact email of the adapter’s maintainer: prebid.support@indexexchange.com

- [x] official adapter submission

### CODE REVIEW

**Comment by dmitriyshashkin** [comment]
Lot's of publishers (us included) said that it's the only reason that  prevents from migrating to 1.X https://github.com/prebid/Prebid.js/issues/2256 I really hope that this will be reviewed as soon as possible :-)

**Comment by Ximich** [comment]
+1, can't wait, so exciting!

**Comment by mercuryyy** [comment]
I tried emailing "prebid.support@indexexchange.com" no response. ran a test with the new build for the last 24-48 hours fill rate is less then a 1/3 in 1.x for index impressions then what is was on 0.x

seems to all be working correctly bids are coming in and rendering but the index side it not winning nearly as much bids as in 0.x

**Comment by ix-prebid-support** [comment]
@mercuryyy 

An endpoint update wasn't completely deployed network wide until 20th April so poor performance was expected during that time frame. Feel free to let us know if you are still seeing an issue.

File: modules/ixBidAdapter.js [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+const BANNER_INSECURE_BID_URL = 'http://as.casalemedia.com/cygnus';
+const SUPPORTED_AD_TYPES = [BANNER];
+const ENDPOINT_VERSION = 7.2;
+const CENT_TO_DOLLAR_FACTOR = 100;
+const TIME_TO_LIVE = 60;
+const NET_REVENUE = true;
+const isSecureWeb = utils.getTopWindowLocation().protocol === 'https:';
+const baseUrl = isSecureWeb ? BANNER_SECURE_BID_URL : BANNER_INSECURE_BID_URL;
+const PRICE_TO_DOLLAR_FACTOR = {
+  JPY: 1
```
Comment (mkendall07) [inline modules/ixBidAdapter.js]: are you intended to hardcode currency rates in your adapter?

**Comment by mercuryyy** [comment]
@ix-prebid-support 
Thanks for the update, issue seems be resolved.

**Comment by jsnellbaker** [comment]
@jas-singh While I was checking over the adapter code I checked-out, I encountered some linting errors while doing a `gulp serve`.

Below is a copy of the errors related to the `ixBidAdapter.js` file:
```
  121:5   error  '&&' should be placed at the end of the line  operator-linebreak
  153:11  error  '&&' should be placed at the end of the line  operator-linebreak
  175:11  error  '||' should be placed at the end of the line  operator-linebreak
```

I'm not sure why these didn't appear in the automated checks, but I think they need to be addressed.  Could you try to rebase off of master again?

**Comment by ix-prebid-support** [comment]
@jsnellbaker 

The lint errors have been resolved

**Comment by akira28** [comment]
So nice that you merged it. Do you plan to release it in 1.9.0? It literally the only thing that prevents us to update to 1.x

**Comment by dmitriyshashkin** [comment]
+1, a release would be much appreciated! We're updating next week anyway, as we need the integration with GDPR consent framework, but it would be better to update to a stable version rather than to the head. Also, thank you for reviewing the PR so fast!

**Comment by Ximich** [comment]
May be release 1.9 Alpha?

File: modules/ixBidAdapter.js [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+
+const BIDDER_CODE = 'ix';
+const BANNER_SECURE_BID_URL = 'https://as-sec.casalemedia.com/cygnus';
+const BANNER_INSECURE_BID_URL = 'http://as.casalemedia.com/cygnus';
+const SUPPORTED_AD_TYPES = [BANNER];
+const ENDPOINT_VERSION = 7.2;
+const CENT_TO_DOLLAR_FACTOR = 100;
+const TIME_TO_LIVE = 60;
+const NET_REVENUE = true;
+const isSecureWeb = utils.getTopWindowLocation().protocol === 'https:';
```
Comment (sumpfork) [inline modules/ixBidAdapter.js]: This line doesn't work for me - `protocol` does not include the colon. Works without the colon.

**Comment by dmitriyshashkin** [comment]
Maybe it would be a good idea to add the alias to the old name (indexExchange). This will make the transition easier for some publishers.

File: modules/ixBidAdapter.js (hunk 2 above)
Comment (ix-prebid-support) [reply modules/ixBidAdapter.js]: This has been addressed in #2496

**Pull request merged**